### PR TITLE
test: add k8s 1.13 to testing framework

### PR DESCRIPTION
**Pull request by aanm**
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6297)

### CODE REVIEW

**Comment by aanm** [comment]
test-missed-k8s

**Comment by aanm** [comment]
test-missed-k8s

**Comment by aanm** [comment]
test-missed-k8s

**Comment by aanm** [comment]
k8s 1.11 failed https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-K8s/1328/execution/node/98/log/ but k8s 1.13 worked fine

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-docs-please

**Comment by aanm** [comment]
test-me-please

File: test/provision/k8s_install.sh [hunk 1]
```diff
@@ -147,6 +174,14 @@ case $K8S_VERSION in
         sudo ln -sf $COREDNS_DEPLOYMENT $DNS_DEPLOYMENT
         KUBEADM_CONFIG="${KUBEADM_CONFIG_ALPHA2}"
         ;;
+    "1.13")
+        KUBERNETES_CNI_VERSION="0.6.0"
+        K8S_FULL_VERSION="1.13.0-beta.2"
+        KUBEADM_OPTIONS="--ignore-preflight-errors=cri"
+        KUBEADM_SLAVE_OPTIONS="--discovery-token-unsafe-skip-ca-verification --ignore-preflight-errors=cri,SystemVerification"
+        sudo ln -sf $COREDNS_DEPLOYMENT $DNS_DEPLOYMENT
```
Comment (eloycoto) [inline test/provision/k8s_install.sh:182]: I think that we should be consistent here, if we run a different version of coredns for 1.12 we also need to run the right version for 1.13. I strongly disagree with this one (We should also use the latest), but if you can add the coredns version for 1.13 will be great.

**Review by eloycoto — CHANGES_REQUESTED** [review]

File: test/provision/k8s_install.sh (hunk 1 above)
Comment (aanm) [reply test/provision/k8s_install.sh:182]: ups I almost forgot. coredns for 1.13 added.

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
k8s 1.10 failed with known flake

**Comment by aanm** [comment]
test-missed-k8s

File: test/provision/manifest/1.13/coredns_deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,189 @@
+# File source
+# https://raw.githubusercontent.com/kubernetes/kubernetes/release-1.12/cluster/addons/dns/coredns/coredns.yaml.base
```
Comment (eloycoto) [inline test/provision/manifest/1.13/coredns_deployment.yaml]: Typo here, is 1.13

**Review by eloycoto — CHANGES_REQUESTED** [review]

**Comment by aanm** [comment]
test-me-please

**Review by eloycoto — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by ianvernon** [comment]
Checkout from git failed: https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/8309/execution/node/7/log/

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
Travis unit tests hit the following issue: 
https://github.com/cilium/cilium/issues/6337

```
level=info msg="Starting to watch allocation changes" kvstoreErr="<nil>" kvstoreStatus="Consul: 172.17.0.3:8300" prefix=test-prefixzdRNAGVhuOFr/id subsys=allocator
allocator_test.go:102:
    c.Assert(id, Equals, idpool.ID(0))
... obtained idpool.ID = 0x1 ("1")
... expected idpool.ID = 0x0 ("0")

START: allocator_test.go:67: AllocatorConsulSuite.TearDownTest
PASS: allocator_test.go:67: AllocatorConsulSuite.TearDownTest	0.006s

FAIL: <autogenerated>:1: AllocatorConsulSuite.TestSelectID

OOPS: 11 passed, 1 FAILED
--- FAIL: Test (28.27s)
FAIL
coverage: 14.5% of statements in ./...
FAIL	github.com/cilium/cilium/pkg/kvstore/allocator	28.347s
make: *** [unit-tests] Error 1
travis_time:end:1253d968:start=1543528581515949791,finish=1543528880499027772,duration=298983077981
[0K[31;1mThe command "./.travis/build.sh" exited with 2.[0m
```

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-missed-k8s

**Comment by aanm** [comment]
test-missed-k8s

**Comment by aanm** [comment]
test-missed-k8s

File: test/provision/k8s_install.sh [hunk 3]
```diff
@@ -178,7 +178,7 @@ case $K8S_VERSION in
         KUBERNETES_CNI_VERSION="0.6.0"
         K8S_FULL_VERSION="1.13.0-beta.2"
         KUBEADM_OPTIONS="--ignore-preflight-errors=cri"
-        KUBEADM_SLAVE_OPTIONS="--discovery-token-unsafe-skip-ca-verification --ignore-preflight-errors=cri"
+        KUBEADM_SLAVE_OPTIONS="--discovery-token-unsafe-skip-ca-verification --ignore-preflight-errors=cri,SystemVerification"
```
Comment (ianvernon) [inline test/provision/k8s_install.sh:181]: Not blocking, just out of curiosity - what is the need to ignore this error?

**Review by ianvernon — APPROVED** [review]

**Comment by ianvernon** [comment]
@aanm looks great overall. Would you find value in documentation for how to update the K8s versions tested in CI so that others can do this in the future as well? I'd be happy to help.

**Comment by aanm** [comment]
test-me-please

File: Documentation/install/upgrade.rst [hunk 4]
```diff
@@ -632,6 +639,11 @@ Download the ConfigMap with the changes for |SCM_BRANCH|
 
       $ wget \ |SCM_WEB|\/examples/kubernetes/1.12/cilium-cm.yaml
 
+  .. group-tab:: K8s 1.12
```
Comment (jrajahalme) [inline Documentation/install/upgrade.rst]: Should be `1.13`

**Review by jrajahalme — APPROVED** [review]
Janitorially looks ok apart from one typo, can't comment on all the k8s yaml detail, though.

File: Documentation/install/upgrade.rst (hunk 4 above)
Comment (aanm) [reply Documentation/install/upgrade.rst]: Correct

**Comment by aanm** [comment]
The tests were all 7/7 green I just made the changes in the documentation pointed out by @jrajahalme

**Comment by aanm** [comment]
test-docs-please

**Pull request merged**